In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from utils import get_data

In [6]:
df_015 = get_data("../data/EE_015/", ["default", "genotype"])
df_050 = get_data("../data/EE_050/", ["default", "genotype"])
df_069 = get_data("../data/EE_069/", ["default", "genotype"])

unvcf_df = pd.concat([df_015, df_050, df_069], ignore_index=True, axis=0)

c:\Users\barte\Desktop\Studies\V semester\pathogenicity-assessment\unvcf_tests\utils.py:43: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  df2 = pd.read_csv(folder_path / f"{sample}_{f_type}.csv.gz", sep=";", compression="gzip").drop("Unnamed: 0", axis=1)


In [7]:
unvcf_df.head(1)

#CHROM    POS         ID REF ALT QUAL FILTER                ACMG_class  \
0   chr1  15820  rs2691315   G   T    .   PASS  Uncertain%40Significance   

  ACMG_coding_impact ACMG_gene  ... ClippingRankSum  ExcessHet  FS MLEAC  \
0       non%40coding         0  ...             NaN        NaN NaN   NaN   

  MLEAF  MQ MQRankSum  QD ReadPosRankSum SOR  
0   NaN NaN       NaN NaN            NaN NaN  

[1 rows x 74 columns]

In [8]:
dt = unvcf_df.dtypes.to_numpy()
cols = unvcf_df.columns
dt_cols = [(d, col) for d, col in zip(dt, cols)]
for d in set(dt):
    print(f"Dtype: {d}:")
    for _d, col in dt_cols:
        if d == _d:
            print(f" - {col}")


Dtype: float64:
 - ACMG_score
 - AMP_score
 - AMP_total_samples
 - ECNT
 - GERMQ
 - MBQ
 - MFRL
 - MMQ
 - MPOS
 - POPAF
 - RPA
 - STR
 - STRQ
 - TLOD
 - cosmicFathMMScore
 - gnomadExomes_AF
 - gnomadExomes_AF_ethnic
 - gnomadGenomes_AF
 - gnomadGenomes_AF_ethnic
 - AC
 - AF
 - AN
 - ExcessHet
 - FS
 - MLEAC
 - MLEAF
 - MQ
 - QD
 - SOR
Dtype: object:
 - #CHROM
 - ID
 - REF
 - ALT
 - QUAL
 - FILTER
 - ACMG_class
 - ACMG_coding_impact
 - ACMG_gene
 - ACMG_rules
 - ACMG_transcript
 - AMP_matches
 - AMP_rules
 - AMP_tier
 - AS_FilterStatus
 - AS_SB_TABLE
 - CGDinheritance
 - ClinVarClass
 - Gene
 - MutationTaster_pred
 - MutationTaster_score
 - RU
 - SIFT_score
 - coding_impact
 - cosmicFathMMPrediction
 - function
 - hgvs
 - BaseQRankSum
 - ClippingRankSum
 - MQRankSum
 - ReadPosRankSum
Dtype: int64:
 - POS
 - ClinVarDisease
 - DANN_score
 - DP
 - gnomadExomesEthnic_AC_Hom
 - gnomadExomes_AC
 - gnomadExomes_AC_Hemi
 - gnomadExomes_AC_Hom
 - gnomadExomes_AN
 - gnomadGenomesAC
 - gnomadGenom

In [9]:
unvcf_df.columns

Index(['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'ACMG_class',
       'ACMG_coding_impact', 'ACMG_gene', 'ACMG_rules', 'ACMG_score',
       'ACMG_transcript', 'AMP_matches', 'AMP_rules', 'AMP_score', 'AMP_tier',
       'AMP_total_samples', 'AS_FilterStatus', 'AS_SB_TABLE', 'CGDinheritance',
       'ClinVarClass', 'ClinVarDisease', 'DANN_score', 'DP', 'ECNT', 'GERMQ',
       'Gene', 'MBQ', 'MFRL', 'MMQ', 'MPOS', 'MutationTaster_pred',
       'MutationTaster_score', 'POPAF', 'RPA', 'RU', 'SIFT_score', 'STR',
       'STRQ', 'TLOD', 'coding_impact', 'cosmicFathMMPrediction',
       'cosmicFathMMScore', 'function', 'gnomadExomesEthnic_AC_Hom',
       'gnomadExomes_AC', 'gnomadExomes_AC_Hemi', 'gnomadExomes_AC_Hom',
       'gnomadExomes_AF', 'gnomadExomes_AF_ethnic', 'gnomadExomes_AN',
       'gnomadGenomesAC', 'gnomadGenomesAN', 'gnomadGenomesEthnic_AC_Hom',
       'gnomadGenomes_AC_Hemi', 'gnomadGenomes_AC_Hom', 'gnomadGenomes_AF',
       'gnomadGenomes_AF_ethnic', 'hgvs', 'AC

In [10]:
all(df_050.columns == df_015.columns)

True

In [11]:
set(df_050.columns).intersection(set(df_069.columns))

{'#CHROM',
 'ACMG_class',
 'ACMG_coding_impact',
 'ACMG_gene',
 'ACMG_rules',
 'ACMG_score',
 'ACMG_transcript',
 'ALT',
 'CGDinheritance',
 'ClinVarClass',
 'ClinVarDisease',
 'DANN_score',
 'DP',
 'FILTER',
 'Gene',
 'ID',
 'MutationTaster_pred',
 'MutationTaster_score',
 'POS',
 'QUAL',
 'REF',
 'SIFT_score',
 'coding_impact',
 'function',
 'gnomadExomesEthnic_AC_Hom',
 'gnomadExomes_AC',
 'gnomadExomes_AC_Hemi',
 'gnomadExomes_AC_Hom',
 'gnomadExomes_AF',
 'gnomadExomes_AF_ethnic',
 'gnomadExomes_AN',
 'gnomadGenomesAC',
 'gnomadGenomesAN',
 'gnomadGenomesEthnic_AC_Hom',
 'gnomadGenomes_AC_Hemi',
 'gnomadGenomes_AC_Hom',
 'gnomadGenomes_AF',
 'gnomadGenomes_AF_ethnic',
 'hgvs'}

In [12]:
set(df_050.columns).difference(set(df_069.columns))

{'AMP_matches',
 'AMP_rules',
 'AMP_score',
 'AMP_tier',
 'AMP_total_samples',
 'AS_FilterStatus',
 'AS_SB_TABLE',
 'ECNT',
 'GERMQ',
 'MBQ',
 'MFRL',
 'MMQ',
 'MPOS',
 'POPAF',
 'RPA',
 'RU',
 'STR',
 'STRQ',
 'TLOD',
 'cosmicFathMMPrediction',
 'cosmicFathMMScore'}

In [13]:
set(df_069.columns).difference(set(df_050.columns))

{'AC',
 'AF',
 'AN',
 'BaseQRankSum',
 'ClippingRankSum',
 'ExcessHet',
 'FS',
 'MLEAC',
 'MLEAF',
 'MQ',
 'MQRankSum',
 'QD',
 'ReadPosRankSum',
 'SOR'}

In [15]:
df_015_csq = get_data("../data/EE_015/", ["default", "csq"])
df_050_csq = get_data("../data/EE_050/", ["default", "csq"])
df_069_csq = get_data("../data/EE_069/", ["default", "csq"])

unvcf_df_csq = pd.concat([df_015_csq, df_050_csq, df_069_csq], ignore_index=True, axis=0)

c:\Users\barte\Desktop\Studies\V semester\pathogenicity-assessment\unvcf_tests\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80) have mixed types. Specify dtype option on import or set low_memory=False.
  df2 = pd.read_csv(folder_path / f"{sample}_{f_type}.csv.gz", sep=";", compression="gzip").drop("Unnamed: 0", axis=1)
c:\Users\barte\Desktop\Studies\V semester\pathogenicity-assessment\unvcf_tests\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80,94) have mixed types. Specify dtype option on import or set low_memory=False.
  df2 = pd.read_csv(folder_path / f"{sample}_{f_type}.csv.gz", sep=";", compression="gzip").drop("Unnamed: 0", axis=1)
c:\Users\barte\Desktop\Studies\V semester\pathogenicity-assessment\unvcf_tests\utils.py:43: DtypeWarning: Columns (26,40,74,75,76,78,80,96) have mixed types. Specify dtype option on import or set low_memory=False.
  df2 = pd.read_csv(folder_path / f"{sample}_{f_type}.csv.gz", sep=";", compression="gzip").drop("Unnamed: 0", axis=1)


In [16]:
df_015_csq

#CHROM       POS           ID REF ALT QUAL FILTER Allele  \
0        chr1     15820    rs2691315   G   T    .   PASS      T   
1        chr1     17385  rs201535981   G   A    .   PASS      A   
2        chr1     17697   rs71260069   G   C    .   PASS      C   
3        chr1    133129  rs367730352   G   A    .   PASS      A   
4        chr1    183629   rs71267774   G   A    .   PASS      A   
...       ...       ...          ...  ..  ..  ...    ...    ...   
102512   chrY  56856512   rs76439603   A   C    .   PASS      C   
102513   chrY  56856517   rs75739861   A   G    .   PASS      G   
102514   chrY  56856554   rs77088808   T   G    .   PASS      G   
102515   chrY  56856581    rs9320083   G   A    .   PASS      A   
102516   chrY  56856587    rs7067511   A   C    .   PASS      C   

                               Consequence    IMPACT  ... SpliceAI_pred_DP_DG  \
0                  downstream_gene_variant  MODIFIER  ...                 NaN   
1                  downstream_gene_variant  MODIFIER  ...                 NaN   
2                  downstream_gene_variant  MODIFIER  ...                 NaN   
3       non_coding_transcript_exon_variant  MODIFIER  ...                 NaN   
4                  downstream_gene_variant  MODIFIER  ...                 NaN   
...                                    ...       ...  ...                 ...   
102512             downstream_gene_variant  MODIFIER  ...                 NaN   
102513             downstream_gene_variant  MODIFIER  ...                 NaN   
102514             downstream_gene_variant  MODIFIER  ...                 NaN   
102515             downstream_gene_variant  MODIFIER  ...                 NaN   
102516             downstream_gene_variant  MODIFIER  ...                 NaN   

       SpliceAI_pred_DP_DL SpliceAI_pred_DS_AG SpliceAI_pred_DS_AL  \
0                      NaN                 NaN                 NaN   
1                      NaN                 NaN                 NaN   
2                      NaN                 NaN                 NaN   
3                      NaN                 NaN                 NaN   
4                      NaN                 NaN                 NaN   
...                    ...                 ...                 ...   
102512                 NaN                 NaN                 NaN   
102513                 NaN                 NaN                 NaN   
102514                 NaN                 NaN                 NaN   
102515                 NaN                 NaN                 NaN   
102516                 NaN                 NaN                 NaN   

       SpliceAI_pred_DS_DG SpliceAI_pred_DS_DL SpliceAI_pred_SYMBOL LOEUF  \
0                      NaN                 NaN                  NaN   NaN   
1                      NaN                 NaN                  NaN   NaN   
2                      NaN                 NaN                  NaN   NaN   
3                      NaN                 NaN                  NaN   NaN   
4                      NaN                 NaN                  NaN   NaN   
...                    ...                 ...                  ...   ...   
102512                 NaN                 NaN                  NaN   NaN   
102513                 NaN                 NaN                  NaN   NaN   
102514                 NaN                 NaN                  NaN   NaN   
102515                 NaN                 NaN                  NaN   NaN   
102516                 NaN                 NaN                  NaN   NaN   

       PHENOTYPES  NMD  
0             NaN  NaN  
1             NaN  NaN  
2             NaN  NaN  
3             NaN  NaN  
4             NaN  NaN  
...           ...  ...  
102512        NaN  NaN  
102513        NaN  NaN  
102514        NaN  NaN  
102515        NaN  NaN  
102516        NaN  NaN  

[102517 rows x 103 columns]

In [17]:
dt = unvcf_df_csq.dtypes.to_numpy()
cols = unvcf_df_csq.columns
dt_cols = [(d, col) for d, col in zip(dt, cols)]
for d in set(dt):
    print(f"Dtype: {d}:")
    for _d, col in dt_cols:
        if d == _d:
            print(f" - {col}")

Dtype: float64:
 - DISTANCE
 - STRAND
 - TSL
 - REFSEQ_MATCH
 - REFSEQ_OFFSET
 - HGVS_OFFSET
 - AF
 - AFR_AF
 - AMR_AF
 - EAS_AF
 - EUR_AF
 - SAS_AF
 - gnomADe_AF
 - gnomADe_AFR_AF
 - gnomADe_AMR_AF
 - gnomADe_ASJ_AF
 - gnomADe_EAS_AF
 - gnomADe_FIN_AF
 - gnomADe_NFE_AF
 - gnomADe_OTH_AF
 - gnomADe_SAS_AF
 - gnomADg_AF
 - gnomADg_AFR_AF
 - gnomADg_AMI_AF
 - gnomADg_AMR_AF
 - gnomADg_ASJ_AF
 - gnomADg_EAS_AF
 - gnomADg_FIN_AF
 - gnomADg_MID_AF
 - gnomADg_NFE_AF
 - gnomADg_OTH_AF
 - gnomADg_SAS_AF
 - MOTIF_POS
 - MOTIF_SCORE_CHANGE
 - EVE_SCORE
 - CADD_PHRED
 - CADD_RAW
 - SpliceAI_pred_DP_AG
 - SpliceAI_pred_DP_AL
 - SpliceAI_pred_DP_DG
 - SpliceAI_pred_DP_DL
 - SpliceAI_pred_DS_AG
 - SpliceAI_pred_DS_AL
 - SpliceAI_pred_DS_DG
 - SpliceAI_pred_DS_DL
 - LOEUF
Dtype: object:
 - #CHROM
 - ID
 - REF
 - ALT
 - QUAL
 - FILTER
 - Allele
 - Consequence
 - IMPACT
 - SYMBOL
 - Gene
 - Feature_type
 - Feature
 - BIOTYPE
 - EXON
 - INTRON
 - HGVSc
 - HGVSp
 - cDNA_position
 - CDS_position
 - Protei

In [18]:
set(df_050_csq.columns) == set(df_015_csq.columns)

True

In [19]:
set(df_050_csq.columns) == set(df_069_csq.columns)

True

In [20]:
set(df_015_csq.columns) == set(df_069_csq.columns)

True

In [70]:
object_columns = set(unvcf_df_csq.select_dtypes(object).columns).difference(set(unvcf_df.select_dtypes(object).columns))
object_columns = list(object_columns)
len(object_columns)

49

In [73]:
for i in range(len(object_columns)):
    print(i)
    print(object_columns[i])
    uniq = unvcf_df_csq[object_columns[i]].unique()
    print(len(uniq))
    print(uniq[:5])
    x = np.argmax([0 if isinstance(y, float) else len(y) for y in uniq])
    print(uniq[x])
    print("========================================")


0
UNIPARC
18096
[nan 'UPI000041820C' 'UPI00001DFBF0' 'UPI000022DAF4' 'UPI00001D7C8B']
UPI000041820C
1
APPRIS
8
[nan 'P1' 'A2' 'P4' 'P3']
P1
2
SYMBOL
23275
['DDX11L1' 'CICP27' 'MIR6859-2' nan 'LINC01409']
PTGER4P2-CDK2AP2P2
3
EXON
2107
[nan '1/1' '12/16' '1/5' '19/19']
107/107
4
Amino_acids
794
[nan 'T' 'E' 'I/V' 'V']
PPMTVTTSGTSQSRSSFSTAKTSTSLHSHTSSTHHPEVTSTSTTSITPNHTST/-
5
SOURCE
3
['Ensembl' nan 'RefSeq']
Ensembl
6
USED_REF
2777
['G' 'C' 'T' 'A' nan]
CTAATCGGCCCATCTCTTTCCCTTTCTGTGTGGTTACCAGGTTACTGGAGGCCACTTAGTTGTTGGTTTTGGTTGGGACTTTCTGGACTTCAGATCTCAGCCCTCGTTTACTCACCCCTGAGCCCTCCCTGCCCTGCCCCCAGCAGTTCCTGACTTCGCCCCTATACACTGTCCCCGGAGCTCTTCCCTGGGC
7
PHENOTYPES
54932
[nan 'Retinitis_pigmentosa+G2P+ENSG00000187634'
 'Retinitis_pigmentosa+G2P+ENSG00000187634&ClinVar:_phenotype_not_specified+ClinVar+rs6672356'
 'Skin_reflectance__Melanin_index_+NHGRI-EBI_GWAS_catalog+rs2272756'
 'Pancreatic_Cancer+NHGRI-EBI_GWAS_catalog+rs13303010']
FASTING_PLASMA_GLUCOSE_LEVEL_QUANTITATIVE_TRAIT_LOCUS_5+MIM_mo

CSQ:
- object columns:
    - UNIPARC - no arrays - 18096 unique
    - APPRIS - no arrays - 8 unique
    - SYMBOL - no arrays - 23275 unique
    - EXON - no arrays - 2107 unique
    - Amino_acids - no arrays - 794 unique
    - SOURCE - no arrays - 3 unique
    - USED_REF - no arays - 2777 unique
    - PHENOTYPES - looks to be an **array** sep=+ ? - _54932 unique_
    - CLIN_SIG - **arrays** sep=& - _184 unique_
    - Allele - no arrays - 2350 unique
    - HGVSc - no arrays - 207584 unique
    - UNIPROT_ISOFORM - no arrays - 9307 unique
    - CCDS - no arrays - 15897 unique
    - ENSP - no arrays - 18398
    - CDS_position - no arrays? (numbers linked xxxx-yyyy) - 6519 unique
    - SOMATIC - **arrays**? (number linked x&x&x&x&y...) - _16 unique_
    - DOMAINS - **arrays** sep=& - _20172 unique_
    - Protein_position - no arrays? (numbers linked xxxx-yyyy) - 3401 unique
    - BIOTYPE - no arrays - 41 unique
    - MANE_PLUS_CLINICAL - no arrays - 17 unique
    - GIVEN_REF - no arrays - 2779 unique
    - PUBMED - **arrays** sep=& - _12572 unique_
    - PolyPhen - no arrays but can be split! into a category and float (probablitity?) e.g. benign(0.737) - _912 unique_
    - HIGH_INF_POS - no arrays - 3 unique
    - CANONICAL - no arrays - 2 unique
    - MANE_SELECT - no arrays - 11361 unique
    - EVE_CLASS - no arrays (nan, uncertain, benign, pathogenic) - 4 unique
    - INTRON - no arrays - 3346 unique
    - TREMBL - **arrays** sep=& - _5451 unique_
    - Consequence - **arrays** sep=& - _94 unique_
    - HGVSp - **arrays**? - a lot of values concatenated (HisHisHisHis...) - _29180 unique_
    - Codons - no arrays lower and uppercase values? - 1430 unique
    - Feature - no arrays - 29671 unique
    - FLAGS - **arrays** sep=& - _5 unique_
    - PHENO -  **arrays**? (number linked x&x&x&x&y...) - _17 unique_
    - SIFT - no arrays but can be split! into a category and float (probablitity?) e.g. tolerated(0.55) - _205 unique_
    - SpliceAI_pred_SYMBOL - no arrays - 15393 unique
    - HGNC_ID - no arrays but can be split? category:number - _22706 unique_
    - Existing_variation - **arrays** sep=& - _268961 unique_
    - SWISSPROT - **arrays** sep=& - _15309 unique_
    - Feature_type - no arrays - 4 unique
    - MOTIF_NAME - no arrays - 8 unique
    - TRANSCRIPTION_FACTORS - **arrays** sep=& - _8 unique_
    - NMD - no arrays (nan, NMD_escaping_variant) - 2 unique
    - SYMBOL_SOURCE - no arrays - 4 unique
    - BAM_EDIT - no arrays (nan, OK) - 2 unique
    - VAR_SYNONYMS - **arrays** specific format sep=& 2 fields separated with :: - _25211 unique_
    - cDNA_position - no arrays? (numbers linked xxxx-yyyy) - _14066 unique_
    - IMPACT - no arrays - 4 unique